### PR TITLE
Add node https proxy support

### PR DESCRIPTION
**Pull request by rmi7**
We (fintech company) would like to use an HTTPS proxy when performing rest calls. Therefore we need to be able to use an HTTPS proxy with this module. So I've added it.

I've tried to follow the `CONTRIBUTING` guidelines (added tests and added the necessary documentation)

If anything is missing or wrong please do tell and I will fix/improve it.

### CODE REVIEW

File: client/node.js [hunk 1]
```diff
@@ -86,6 +87,10 @@ module.exports = client(function node (request) {
       reject(response)
     }
 
+    if (request.agent) {
```
Comment (scothis) [inline client/node.js]: It's not necessary to add a new property to the request to set the agent. You can currently define the agent via `request.mixin.agent`.

**Review by scothis — COMMENTED** [review]
@rmi7 thanks for the contribution. This is a great example of how an interceptor can add significant new functionality.

At this time, I'm hesitant to introduce a new dependency for the project. Zero dependencies is a magical place to be. That said, there is plenty of room for third party interceptors. I'd be happy to add a link to your package in the documentation.

File: client/node.js (hunk 1 above)
Comment (rmi7) [inline client/node.js]: thanks for the tip and quick response!

that does make it even nicer, not having to alter `client/node.js`.

I'm working on a separate module to be used as a plugin, will update here once it's done on monday.

**Comment by rmi7** [comment]
I've created a plugin module [rest-node-https-proxy-interceptor](https://github.com/dimebox/rest-node-https-proxy-interceptor) (also published it on npm).

please take a look and if your happy with it place a reference in the documentation, otherwise let me know what still needs to be improved.

thanks for the quick response on this PR

**Comment by scothis** [comment]
rest.js is no longer maintained

**Pull request closed without merging**